pkg/config/config.go

```diff
@@ -215,6 +216,33 @@ func (c *ChartConfig) GetFormattedKeyWithDefaultValue(xc *XPathConfig, prefix st
 	return key, keyType
 }
 
+func (c *ChartConfig) Validate() error {
+	// Currently validate that file-if cannot be present at root level file configs
+	// and that globalConfig cannot contain a root level entry
+	if _, ok := c.GlobalConfig[XPathRoot]; ok {
+		return fmt.Errorf("cannot have root level config in GlobalConfig")
+	}
+
+	for manifest, config := range c.FileConfig {
+		for xpath, xpathConfigs := range config {
+			for _, xpathConfig := range xpathConfigs {
+				strategy := xpathConfig.Strategy
+				if xpath == XPathRoot {
+					if strategy != XPathStrategyFileIf {
+						// with or range might be valid here one day to create multiple
+						// instances of the same kind with slightly different properties.
+						return fmt.Errorf("'%s' cannot use strategy '%s' at root level", manifest, strategy)
+					}
+				} else if strategy == XPathStrategyFileIf {
+					return fmt.Errorf("'%s' cannot use strategy '%s' at non-root level", manifest, strategy)
+				}
+			}
+		}
+	}
+
+	return nil
```

Comment:
These could be simplified to 
```golang
if xpath != XPathRoot || strategy != XPathStrategyFileIf {
    return fmt.Errorf(...)
}
```

Reply:
Removed one level of if nesting, but not the way you suggested as that causes several test failures.

Reply:
Sorry, please see the latest commit. `make test` reports no error.

Reply:
Ahh - I see. 

What you've actually committed is different to what you put in the comment above, but a neat trick though.